auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java

```diff
@@ -329,6 +348,22 @@ public boolean checkRight(String actorCrn, String userCrn, String right, Optiona
         return checkRight(actorCrn, userCrn, right, null, requestId);
     }
 
+    /**
+     * Retrieves whether the member has the specified rights.
+     *
+     * @param actorCrn the CRN of the actor
+     * @param memberCrn the CRN of the member
+     * @param rightChecks the rights to check
+     * @param requestId an optional request id
+     * @return a list of booleans indicating whether the member has the specified rights
+     */
+    public List<Boolean> hasRights(String actorCrn, String memberCrn, List<AuthorizationProto.RightCheck> rightChecks, Optional<String> requestId) {
```

Comment:
please check checkRight method and add caching to this method similarly

Reply:
Caching is not a good idea for my use case. Here's a bad scenario that may occur with caching.
1. User access is removed
2. FMS checks event generation ids, sees that they have changed, and runs user sync
3. user sync gets the cached value for hasRights that still grants user access to environment
4. FMS saves the event generation ids for successful sync and thinks that all users are up-to-date
In this case, a user may retain access to an environment that they should not have access to.

Reply:
Our caches has small expiration time (1 sec afaik), so if permissions change, then it will turn out quickly and it prevents cb to overload network with ums calls, that is the reason why we have caches. If you think 1 sec expiration time still can cause issue, then we can skip cache here, but we had performance issues with UMS on mow-dev and/or mow-int.

Reply:
I think the performance issues you had/have w/ checkRight is because the authorization layer is down at the database access level. From my reading, each database access, either read or write, requires a checkRight call.

The use-case here is different from protecting database access. Load here is determined by the application layer. I'm explicitly calling hasRights once per user/machineUser per user sync. The problem with caching isn't the expiration time, but that the rights check has been decoupled from other activities in the UMS or AuthorizationService. Decoupling can result in incorrect behavior and lead to a security hole. I think this is especially dangerous because the caching is invisible to the caller of hasRights and changes to the caching behavior may be made behind the scenes without understanding the callers.

Reply:
I see your point, alright, then we should go with this.